### DISCUSSION TITLE
Edge Functions Custom Domains

### DISCUSSION THREAD

**Discussion by leekuo** [supabase/supabase#13371] [Feature Requests] [answered]
I saw this link: https://supabase.com/docs/guides/platform/custom-domains#limitations

"Edge functions do not honor the custom domain or the vanity subdomain setting and they still have to be invoked via the foobarbaz.supabase.co domain."

I have created several edge functions but I want to use my own custom domain.  As there are many clients still pointing at my own domain, I don't have the option of switching them all over.  Is there anyway to enable custom domains for edge functions somehow?  Or if not, is there somewhere else to host these deno functions where I can do so?

**Comment by codingcronus** [comment]
I have the same issue. Looking forward to a solution, Supabase.

**Comment by haschu** [comment]
Same here! That feature would be great - it's already supported by Deno Deploy:
https://deno.com/deploy/docs/custom-domains

**Comment by bouwermike** [comment]
Same issue - I don't like that the Edge function URL exposes my project's ref. I want to customise this please!

**Comment by raf-avag** [comment]
any news on this?

**Comment by arioberek** [comment]
Have there been any updates regarding this?

**Reply by valpackett** [reply]
Well, seems like the "limitations" section is gone from that docs page, and instead it lists

> You are creating APIs for third-party systems, for example, implementing webhooks or external API calls to your project via [Edge Functions](https://supabase.com/docs/guides/functions).

as a reason to use a custom domain…

**Comment by inian** [comment, marked as answer]
Custom domains are supported on edge functions.

**Reply by valscholz** [reply]
@inian I've set up custom domain on my Supabase project, but the environment variables for supabase_url are still going on the old supabase domain. This breaks my web app as publicUrls generated in the edge function link to the wrong domain. How can we update the environment variables for edge functions after activating the custom domain in Supabase?

**Reply by argshook** [reply]
having the same issue.

after setting up custom domain, i can call my edge function via custom
domain, but `SUPABASE_URL` env variable is pointing to the original supabase project URL and not to the custom domain.

i'm unable to change `SUPABASE_URL` env variable, due to error:

```sh
$ supabase secrets set SUPABASE_URL="https://api.my-new-domain.com"
Env name cannot start with SUPABASE_, skipping: SUPABASE_URL
Try rerunning the command with --debug to troubleshoot the error.
```

I'm using edge function to upload files to supabase storage and respond with the uploaded URL with:

```ts
const url = supabase.storage.from('my-bucket-name').getPublicUrl('my-file').data.publicUrl
```

the `url` is pointing to `https://abcdefgh.supabase.co/storage...` instead of `https://api.my-new-domain.com/storage...`.

i can fix this on my own by replacing the domain, still would be nice to avoid such juggling.
